encoder/libav_encoder.cpp

```diff
@@ -656,13 +722,32 @@ void LibAvEncoder::audioThread()
 			AVRational num = { 1, out_frame->sample_rate };
 			int64_t ts = av_rescale_q(audio_samples_, num, codec_ctx_[AudioOut]->time_base);
 
-			out_frame->pts = ts + (options_->av_sync > 0 ? options_->av_sync : 0);
-			audio_samples_ += codec_ctx_[AudioOut]->frame_size;
+			// Need to set up timestamps based on the video recording settings:
+			if (previous_audio_timestamp_ == 0)
+				previous_audio_timestamp_ = ts; // Only useful on startup
 
-			ret = avcodec_send_frame(codec_ctx_[AudioOut], out_frame);
-			if (ret < 0)
-				throw std::runtime_error("libav: error encoding frame: " + std::to_string(ret));
+			int64_t et = ts - previous_audio_timestamp_; // Elapsed Time
 
+			if (feed_encoder_frames_ && options_->keypress)
+			{
+				// If feeding audio and keypress option is on, then count the frame timer for the audio frames
+				virtual_audio_ts_ += et;
+				out_frame->pts = virtual_audio_ts_;
+			}
+			else
+				out_frame->pts = ts - virtual_audio_ts_ + (options_->av_sync < 0 ? -options_->av_sync : 0);
+
+			previous_audio_timestamp_ = ts; // Previous TimeStamp
+			audio_samples_ += codec_ctx_[AudioOut]->frame_size;
+			{
+				std::scoped_lock<std::mutex> lock(reset_mutex_);
```

Comment:
Just wondering about the placement of the lock. In the video thread, the lock covers the call to `encode()` but here it doesn't. Is one of them incorrect maybe?

Reply:
I think it ought to *not* cover ``encode()`` in both cases.  But we ought to test this thoroughly.

Reply:
Have removed mutexes. Seems to record very long videos with lots of segments without issue. Have confirmed audio also works.